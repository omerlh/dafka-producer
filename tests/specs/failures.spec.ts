import type {Orchestrator} from '../testcontainers/orchestrator.js';
import {start as startKafka} from '../testcontainers/orchestrator.js';

describe('tests', () => {
    let orchestrator: Orchestrator;

    beforeAll(async () => {
        orchestrator = await startKafka({
            KAFKA_AUTO_CREATE_TOPICS_ENABLE: 'false',
        });
    }, 1800000);

    afterAll(async () => {
        await orchestrator.stop();
    }, 1800000);

    it('should throw if failed to produce', async () => {
        const topic = `topic-${Date.now()}`;

        expect(() =>
            orchestrator.produce([
                {
                    topic,
                    value: {data: 'foo'},
                },
            ])
        ).rejects.toMatchSnapshot();
    });
});
