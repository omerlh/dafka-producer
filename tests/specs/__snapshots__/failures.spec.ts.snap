// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests should throw if failed to produce 1`] = `[TypeError: fetch failed]`;
